maven-surefire-common/src/test/java/org/apache/maven/plugin/surefire/report/StatelessXmlReporterTest.java

```diff
@@ -380,7 +381,7 @@ public void testSyncOnDeferredFile() throws Exception
     {
         Utf8RecodingDeferredFileOutputStream out = new Utf8RecodingDeferredFileOutputStream( "test" );
         ByteBuffer cache = ByteBuffer.wrap( new byte[] {1, 2, 3} );
-        cache.position( 3 );
+        ( (Buffer) cache ).position( 3 );
```

Comment:
Can we declare the `cache` as `Buffer` in the tests as well?

Reply:
I checked all tests and also all application code by manually flipping local `ByteBuffer` variables to `Buffer`, and this is the only place where this is actually possible. In all other places one of the following stops me from easily changing  `ByteBuffer` into `Buffer` types:
  * Other `ByteBuffer` methods are being used, requiring casts in the reverse direction to `ByteBuffer`.
  * The variable is declared as a method parameter, requiring signature changes, possibly recursively.
  * The variable is used in method calls requiring a cast in reverse direction to `ByteBuffer`.

Reply:
Sorry for kicking off all CI builds for this little change. Thanks for the review.